lib/elixir/test/elixir/task_test.exs

```diff
@@ -272,6 +272,139 @@ defmodule TaskTest do
     end
   end
 
+  @tag timeout: 10
+  describe "await_many/2" do
```

Comment:
The tests use "fake tasks". Let's avoid doing that. Please start tasks instead of defining refs by hand.

Reply:
If you need a task that never returns, you can do this: `Task.async(fn -> Process.sleep(:infinity) end)`.

Reply:
How do you recommend I control the order in which the tasks send replies for this first test? I'd like to ensure that the return value is ordered even if the replies are sent out of order. I could give each task a tiny `sleep`, but that feels a little hacky?

Reply:
Heh, and likewise switching to real tasks is giving me trouble in `test "ignores messages from other processes"` because the fake response now arrives before the real one. A `sleep 1` before sending the fake response solves it for me, but I'm not sure that's very robust.

Reply:
Ah, for the test that asserts the ordering and for the fake messages in particular, it is ok to use fake messages. But we also need a test that works directly on the tasks. :)